Day0/krischal111/assignment2/solution.c

```diff
@@ -0,0 +1,104 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+#define expressionfile "expressions.txt"
+#define resultfile "result.txt"
+
+int getnum(FILE * myfile)
+{
+    char numberstring[10] = "         ";
+    
+    // removing whitespace at the beginning
+    char c = ' ';
+    while(c == ' ')
+        c = fgetc(myfile);
+        
+    // extracting number as string into a variable
+    for(int i = 0; i<10; i++)
+    {
+        numberstring[i] = c;
+        
+        // break when numbers are completed
+        if(c < '0' || c > '9')
+        {
+            fseek(myfile, -1, 1); // Ek step paxadi aija vaneko
+            break;
+        }
+        c = fgetc(myfile);
+    }
+    
+    // return the number in our string
+    return atoi(numberstring);
+}
+
+char getoperator(FILE* myfile)
+{
+    char c;
+    do
+    {
+        c = fgetc(myfile);
+    } while (c == ' ') ;
+    
+    return c;
+}
+
+int main()
+{
+    FILE * myfile = fopen(expressionfile, "r");
```

Comment:
Recommended to Read/Open file as binary i.e. "rb" and not "r" always, same for writing. More on that here: https://stackoverflow.com/questions/229924/difference-between-files-written-in-binary-and-text-mode

Reply:
Okay. Didn't know that reading text in binary mode was okay, but not vice versa. I've corrected that now.